SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp

```diff
@@ -837,12 +837,13 @@ ERPCResult USpatialSender::SendRPCInternal(UObject* TargetObject, UFunction* Fun
 		if (SpatialGDKSettings->UseRPCRingBuffer() && RPCService != nullptr)
 		{
 			EPushRPCResult Result = RPCService->PushRPC(TargetObjectRef.Entity, RPCInfo.Type, Payload);
-#if !UE_BUILD_SHIPPING
 			if (Result == EPushRPCResult::Success || Result == EPushRPCResult::QueueOverflowed)
 			{
+				FlushRPCService();
```

Comment:
We shouldn't flush when it's overflowed (or rather there's no point flushing then), so I prefer this living in a separate if.